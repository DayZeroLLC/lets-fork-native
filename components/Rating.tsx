import React from 'react'
import { View, StyleSheet } from 'react-native'
import { MaterialIcons } from '@expo/vector-icons'

type StarProps = {
  fill: 'full' | 'half';
}

function Star(props: StarProps): React.ReactElement {
  const { fill } = props

  return (
    <View style={fill === 'half' && styles.half}>
      <MaterialIcons styles={styles.half} name="star" size={32} color="white" />
    </View>
  )
}


type RatingProps = {
  rating: number;
}

export default function Rating(props: RatingProps): React.ReactElement {
  const { rating } = props

  const stars = [...Array(Math.floor(rating))].map((e, i: number) => (
    <Star key={i} fill="full" /> // eslint-disable-line react/no-array-index-key
  ))

  if (rating - Math.floor(rating) > 0) {
    stars.push(<Star key={0.5} fill="half" />)
  }

  return (
    <View style={styles.rating}>
      {stars}
    </View>
  )
}

const styles = StyleSheet.create({
  half: {
    width: 16,
  },
  rating: {
    flexDirection: 'row',
  },
})
