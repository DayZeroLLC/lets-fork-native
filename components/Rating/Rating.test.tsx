import React from 'react'
import { render } from '@testing-library/react-native'
import Rating from 'components/Rating'
import { toMatchDiffSnapshot } from 'snapshot-diff'

expect.extend({ toMatchDiffSnapshot })

test('Rating', () => {
  const { rerender, asJSON } = render(
    <Rating size="sm" rating={4} />,
  )

  const firstRender = asJSON()

  rerender(
    <Rating size="lg" rating={4} />,
  )

  expect(firstRender).toMatchDiffSnapshot(asJSON())

  rerender(
    <Rating size="sm" rating={3} />,
  )

  expect(firstRender).toMatchDiffSnapshot(asJSON())
})
