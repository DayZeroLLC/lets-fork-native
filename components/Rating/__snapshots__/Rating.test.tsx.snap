// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -12,11 +12,11 @@
      style={
        Object {
          \\"marginBottom\\": -28,
          \\"marginTop\\": -28,
          \\"resizeMode\\": \\"contain\\",
-         \\"width\\": 80,
+         \\"width\\": 160,
        }
      }
      testID=\\"rating\\"
    />
  </View>"
`;

exports[`Rating 2`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
